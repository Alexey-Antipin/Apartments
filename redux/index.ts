import { checkbox, toogleBox, reset } from "./reducers/checkboxReducer";
import { newsThunk, countNewsThunk } from "./reducers/newsReducer";
import { amountOfRoomsThunk } from "./reducers/mainReducer";
import { articlesThunk } from "./reducers/articlesReducer";
import { choiceCity } from "./reducers/catalogReducer";
import {
  authorizationThunk,
  accountUser,
  accountDelete,
} from "./reducers/authorizationReducer";
import {
  redistrationThunk,
  confirmDataThunk,
  toggleCaptcha,
  captchaThunk,
} from "./reducers/registrationReducer";
import {
  contactsThunk,
  closeModal,
  openModal,
} from "./reducers/contactsReducer";
import {
  selectMetroMainPage,
  selectAreaMainPage,
  selectCountRooms,
  selectCallCity,
  selectPriceMin,
  selectPriceMax,
  selectPlaces,
  defaultPrice,
  selectMetro,
  selectCity,
  selectArea,
} from "./reducers/selectReducer";

import { useAppDispatch, useAppSelector } from "../redux/hooks";
import type { RootState, AppDispatch } from "./store";

export {
  // Thunk
  authorizationThunk,
  amountOfRoomsThunk,
  redistrationThunk,
  confirmDataThunk,
  countNewsThunk,
  articlesThunk,
  contactsThunk,
  captchaThunk,
  newsThunk,

  // Method
  selectMetroMainPage,
  selectAreaMainPage,
  selectCountRooms,
  selectPriceMin,
  selectPriceMax,
  selectCallCity,
  accountDelete,
  toggleCaptcha,
  selectPlaces,
  defaultPrice,
  selectMetro,
  accountUser,
  selectCity,
  selectArea,
  choiceCity,
  closeModal,
  toogleBox,
  openModal,
  checkbox,
  reset,

  // Hooks
  useAppDispatch,
  useAppSelector,

  //Type Store
  AppDispatch,
  RootState,
};
