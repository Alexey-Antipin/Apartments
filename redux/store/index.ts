import { configureStore } from "@reduxjs/toolkit";
import authorizationSlice from "../reducers/authorizationReducer";
import registrationSlice from "../reducers/registrationReducer";
import contactsSlice from "../reducers/contactsReducer";
import newsSlice from "../reducers/newsReducer";

export const store = configureStore({
  reducer: {
    authorization: authorizationSlice,
    registration: registrationSlice,
    contacts: contactsSlice,
    news: newsSlice,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
