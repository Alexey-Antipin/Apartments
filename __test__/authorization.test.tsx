import Authorization from "../pages/authorization";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../redux/store";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "./",
      pathname: "",
      query: "",
      asPath: "/",
      push: jest.fn(),
    };
  },
}));

describe("Authorization", () => {
  test("Head", () => {
    const { getByText } = render(
      <Provider store={store}>
        <Authorization />
      </Provider>
    );
    let head = getByText(/Авторизация/i);
    expect(head).toBeInTheDocument();
  });
  test("button", () => {
    const { getByRole } = render(
      <Provider store={store}>
        <Authorization />
      </Provider>
    );
    let button = getByRole("button", { name: /Войти/i });
    expect(button).toBeInTheDocument();
  });
  test("component snapshot", () => {
    const component = render(
      <Provider store={store}>
        <Authorization />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});