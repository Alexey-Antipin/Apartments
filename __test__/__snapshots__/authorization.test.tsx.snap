// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorization component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="background"
      >
        <div
          class="container"
        >
          <div
            class="authorization"
          >
            <h1
              class="title"
            >
              Авторизация
            </h1>
            <p
              class="paragraph"
            >
              Авторизируйтесь, чтобы начать публиковать свои объявления
            </p>
            <form
              action="#"
            >
              <label
                class="position"
                for="login"
              >
                <input
                  class="input"
                  id="login"
                  name="login"
                  placeholder="Логин"
                  value=""
                />
                <span
                  class="icons"
                >
                  <svg
                    fill="#686868"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      clip-path="url(#clip0_305_638)"
                    >
                      <path
                        d="M10.0013 0C7.14418 0 4.80859 2.33559 4.80859 5.19275C4.80859 8.04991 7.14418 10.3855 10.0013 10.3855C12.8585 10.3855 15.1941 8.04991 15.1941 5.19275C15.1941 2.33559 12.8585 0 10.0013 0Z"
                      />
                      <path
                        d="M18.913 14.535C18.7769 14.1949 18.5955 13.8774 18.3915 13.5826C17.3484 12.0407 15.7384 11.0203 13.9244 10.7708C13.6976 10.7482 13.4482 10.7935 13.2668 10.9296C12.3144 11.6325 11.1806 11.9953 10.0014 11.9953C8.82228 11.9953 7.68851 11.6325 6.73612 10.9296C6.5547 10.7935 6.30526 10.7255 6.07853 10.7708C4.26446 11.0203 2.63183 12.0407 1.61143 13.5826C1.40735 13.8774 1.22592 14.2176 1.0899 14.535C1.02189 14.6711 1.04454 14.8298 1.11256 14.9659C1.29398 15.2833 1.52071 15.6008 1.72479 15.8729C2.04224 16.3038 2.38239 16.6892 2.76789 17.052C3.08534 17.3695 3.44815 17.6643 3.81099 17.9591C5.60236 19.297 7.75657 19.9999 9.97879 19.9999C12.201 19.9999 14.3552 19.2969 16.1466 17.9591C16.5094 17.687 16.8722 17.3695 17.1897 17.052C17.5525 16.6892 17.9153 16.3037 18.2328 15.8729C18.4595 15.5781 18.6636 15.2833 18.845 14.9659C18.9583 14.8298 18.981 14.6711 18.913 14.535Z"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="clip0_305_638"
                      >
                        <rect
                          fill="white"
                          height="20"
                          width="20"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </span>
              </label>
              <label
                class="position"
                for="password"
              >
                <input
                  class="input"
                  id="password"
                  name="password"
                  placeholder="Пароль"
                  type="password"
                  value=""
                />
                <span
                  class="icons"
                >
                  <svg
                    fill="#686868"
                    height="20"
                    viewBox="0 0 16 20"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.625 7.5H13V5C13 2.2425 10.7575 0 7.99999 0C5.24249 0 2.99999 2.2425 2.99999 5V7.5H2.37499C1.34166 7.5 0.499992 8.34083 0.499992 9.375V18.125C0.499992 19.1592 1.34166 20 2.37499 20H13.625C14.6583 20 15.5 19.1592 15.5 18.125V9.375C15.5 8.34083 14.6583 7.5 13.625 7.5ZM4.66666 5C4.66666 3.16167 6.16166 1.66667 7.99999 1.66667C9.83833 1.66667 11.3333 3.16167 11.3333 5V7.5H4.66666V5ZM8.83333 13.935V15.8333C8.83333 16.2933 8.46083 16.6667 7.99999 16.6667C7.53916 16.6667 7.16666 16.2933 7.16666 15.8333V13.935C6.67083 13.6458 6.33333 13.1142 6.33333 12.5C6.33333 11.5808 7.08083 10.8333 7.99999 10.8333C8.91916 10.8333 9.66666 11.5808 9.66666 12.5C9.66666 13.1142 9.32916 13.6458 8.83333 13.935Z"
                    />
                  </svg>
                </span>
              </label>
              <div
                class="control"
              >
                <label
                  class="label-block"
                  for="switch"
                >
                  <input
                    class="switch-block"
                    id="switch"
                    name="remember"
                    type="checkbox"
                    value="false"
                  />
                  <div
                    class="switch"
                  >
                    <div
                      class="round"
                    />
                  </div>
                  <div
                    class="remember"
                  >
                    Запомнить меня
                  </div>
                </label>
                <a
                  class="forgot"
                  href="/"
                >
                  Забыли пароль?
                </a>
              </div>
              <button
                class="entrance"
                type="submit"
              >
                Войти
              </button>
            </form>
            <p
              class="account"
            >
              Еще нет аккаунта?
              <a
                class="account-create"
                href="/registration"
              >
                Создайте акканут
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="background"
    >
      <div
        class="container"
      >
        <div
          class="authorization"
        >
          <h1
            class="title"
          >
            Авторизация
          </h1>
          <p
            class="paragraph"
          >
            Авторизируйтесь, чтобы начать публиковать свои объявления
          </p>
          <form
            action="#"
          >
            <label
              class="position"
              for="login"
            >
              <input
                class="input"
                id="login"
                name="login"
                placeholder="Логин"
                value=""
              />
              <span
                class="icons"
              >
                <svg
                  fill="#686868"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_305_638)"
                  >
                    <path
                      d="M10.0013 0C7.14418 0 4.80859 2.33559 4.80859 5.19275C4.80859 8.04991 7.14418 10.3855 10.0013 10.3855C12.8585 10.3855 15.1941 8.04991 15.1941 5.19275C15.1941 2.33559 12.8585 0 10.0013 0Z"
                    />
                    <path
                      d="M18.913 14.535C18.7769 14.1949 18.5955 13.8774 18.3915 13.5826C17.3484 12.0407 15.7384 11.0203 13.9244 10.7708C13.6976 10.7482 13.4482 10.7935 13.2668 10.9296C12.3144 11.6325 11.1806 11.9953 10.0014 11.9953C8.82228 11.9953 7.68851 11.6325 6.73612 10.9296C6.5547 10.7935 6.30526 10.7255 6.07853 10.7708C4.26446 11.0203 2.63183 12.0407 1.61143 13.5826C1.40735 13.8774 1.22592 14.2176 1.0899 14.535C1.02189 14.6711 1.04454 14.8298 1.11256 14.9659C1.29398 15.2833 1.52071 15.6008 1.72479 15.8729C2.04224 16.3038 2.38239 16.6892 2.76789 17.052C3.08534 17.3695 3.44815 17.6643 3.81099 17.9591C5.60236 19.297 7.75657 19.9999 9.97879 19.9999C12.201 19.9999 14.3552 19.2969 16.1466 17.9591C16.5094 17.687 16.8722 17.3695 17.1897 17.052C17.5525 16.6892 17.9153 16.3037 18.2328 15.8729C18.4595 15.5781 18.6636 15.2833 18.845 14.9659C18.9583 14.8298 18.981 14.6711 18.913 14.535Z"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_305_638"
                    >
                      <rect
                        fill="white"
                        height="20"
                        width="20"
                      />
                    </clippath>
                  </defs>
                </svg>
              </span>
            </label>
            <label
              class="position"
              for="password"
            >
              <input
                class="input"
                id="password"
                name="password"
                placeholder="Пароль"
                type="password"
                value=""
              />
              <span
                class="icons"
              >
                <svg
                  fill="#686868"
                  height="20"
                  viewBox="0 0 16 20"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.625 7.5H13V5C13 2.2425 10.7575 0 7.99999 0C5.24249 0 2.99999 2.2425 2.99999 5V7.5H2.37499C1.34166 7.5 0.499992 8.34083 0.499992 9.375V18.125C0.499992 19.1592 1.34166 20 2.37499 20H13.625C14.6583 20 15.5 19.1592 15.5 18.125V9.375C15.5 8.34083 14.6583 7.5 13.625 7.5ZM4.66666 5C4.66666 3.16167 6.16166 1.66667 7.99999 1.66667C9.83833 1.66667 11.3333 3.16167 11.3333 5V7.5H4.66666V5ZM8.83333 13.935V15.8333C8.83333 16.2933 8.46083 16.6667 7.99999 16.6667C7.53916 16.6667 7.16666 16.2933 7.16666 15.8333V13.935C6.67083 13.6458 6.33333 13.1142 6.33333 12.5C6.33333 11.5808 7.08083 10.8333 7.99999 10.8333C8.91916 10.8333 9.66666 11.5808 9.66666 12.5C9.66666 13.1142 9.32916 13.6458 8.83333 13.935Z"
                  />
                </svg>
              </span>
            </label>
            <div
              class="control"
            >
              <label
                class="label-block"
                for="switch"
              >
                <input
                  class="switch-block"
                  id="switch"
                  name="remember"
                  type="checkbox"
                  value="false"
                />
                <div
                  class="switch"
                >
                  <div
                    class="round"
                  />
                </div>
                <div
                  class="remember"
                >
                  Запомнить меня
                </div>
              </label>
              <a
                class="forgot"
                href="/"
              >
                Забыли пароль?
              </a>
            </div>
            <button
              class="entrance"
              type="submit"
            >
              Войти
            </button>
          </form>
          <p
            class="account"
          >
            Еще нет аккаунта?
            <a
              class="account-create"
              href="/registration"
            >
              Создайте акканут
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
